source/extensions/quic_listeners/quiche/BUILD

```diff
@@ -367,6 +365,32 @@ envoy_cc_extension(
     ],
 )
 
+# Create a single target that contains all the libraries that register factories.
+# All of these are needed for this extension to function.
+envoy_cc_extension(
+    name = "quic_factory_lib",
+    security_posture = "unknown",
+    tags = ["nofips"],
+
+    # QUICHE can't build against FIPS BoringSSL until the FIPS build
+    # is on a new enough version to have QUIC support. Remove it from
+    # the build until then. Re-enable as part of #7433.
+    #
+    # QUICHE doesn't compile on Windows right now. Disable until it
+    # is fixed.
+    deps = select({
+        "//bazel:boringssl_fips": [],
+        "//bazel:boringssl_disabled": [],
+        "//bazel:windows_x86_64": [],
+        "//conditions:default": [
+            ":active_quic_listener_config_lib",
+            ":codec_lib",
+            ":quic_transport_socket_factory_lib",
+            ":udp_gso_batch_writer_config_lib",
```

Comment:
I wouldn't put GSO library into the default QUICHE extensions.

Reply:
Would you want a separately compilable extension just for GSO with quiche? In general, we include all features by default. Given that this one is only turned on with specific config, it seems like it should be built by default.

Reply:
GSO is not a necessary for using QUICHE, and it can also be used by other UDP listener. I would prefer to make GSO a stand alone extension.

Reply:
+1. I asked for this as part of the original GSO review.

Reply:
So is the plan to do this in a follow up? Or as part of this PR?

Reply:
I'd rather defer that to a future PR.

Currently `UdpGsoBatchWriter` inherits from `quic::QuicGsoBatchWriter`. I haven't looked into it at all, but does that mean it either requires QUICHE for it's implementation, or only works for quic?

Reply:
> I haven't looked into it at all, but does that mean it either requires QUICHE for it's implementation, or only works for quic?

There is a thread on this in the original PR where I commented that we should not depend on QUICHE for this, since it's applicable to the normal UDP proxy case. I recommend just filing a follow up issue here to untangle this?

Reply:
It depends on QUICHE because the GSO writer is in that extension, but it doesn't depends on other QUIC objects, e.g. stream, connection...
UdpGsoBatchWriter can work for other listeners not just QUIC.

Reply:
Filed #12921